asset-transfer-secured-agreement/application-gateway-typescript/src/contractWrapper.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Contract, ProposalOptions } from '@hyperledger/fabric-gateway';
+import { TextDecoder } from 'util';
+
+export class ContractWrapper {
```

Comment:
Right now, this class is adding so little to the Fabric Gateway client API, I suspect it would be better not to have the extra code to look through. When talking about providing a more business-level abstraction of the smart contract at the client end, I was thinking something like this:

https://github.com/hyperledger/fabric-samples/blob/05791d30bc68fee8f9408e34bc1d5cee68d842f9/off_chain_data/application-typescript/src/contract.ts

So having a class with methods on it that correspond to the smart contract transaction functions (or combinations of transaction functions) used by the client application, but which hide the mechanics of invoking those transactions, and un/marshalling both the arguments and responses.

With that in place, the main body of the application _might_ work as a flow of calls to that contract abstraction and not require all the the specific methods (like createAsset, readAsset etc.) in the app.ts code itself.

Reply:
What you have now looks much better. I can't seem to mark this resolved so just a note to ignore it now.